src/common/ittnotify/ittnotify_static.c

```diff
@@ -62,7 +62,7 @@ static const char api_version[] = API_VERSION "\0\n@(#) $Revision$\n";
 #define ITT_ATTRIBUTE_FALLTHROUGH [[gnu::fallthrough]]
 #elif HAS_CPP_ATTR(clang::fallthrough)
 #define ITT_ATTRIBUTE_FALLTHROUGH [[clang::fallthrough]]
-#elif HAS_GNU_ATTR(fallthrough) && !defined(__INTEL_COMPILER)
+#elif HAS_GNU_ATTR(fallthrough) && !__INTEL_COMPILER
 #define ITT_ATTRIBUTE_FALLTHROUGH __attribute__((fallthrough))
```

Comment:
__INTEL_COMPILER is a macro which is only defined as compiler version when intel compiler is used, so seems `!defined`  is better? 
https://stackoverflow.com/questions/13534032/how-to-detect-intels-compiler-icc-with-ifdef

Reply:
I tested on VC2019 and GCC, as long as __INTEL_COMPILER is not defined as 0, `!__INTEL_COMPILER` is the same as `!defined(__INTEL_COMPILER)` in practice, so I guess we can keep that

Reply:
This is oneDNN custom change (haven't been promoted to ITT) which comes as warning if `-Wundef` is used for compilation.

Reply:
Oh, I see, I also double confirmed it by checking all references to `__INTEL_COMPILER` in oneDNN master, all of them are `defined(__INTEL_COMPILER)` except this one, I guess it was actually changed by OpenVINO guys from `!__INTEL_COMPILER` to `defined(__INTEL_COMPILER)`  in our fork for a reason?